samples/net/cloud/aws_iot_mqtt/Kconfig

```diff
@@ -73,6 +73,12 @@ config AWS_EXPONENTIAL_BACKOFF
 	help
 	  Enable AWS exponential backoff for reconnecting to AWS MQTT broker.
 
+
+config AWS_USE_MQTT_OVER_PORT_443
+	bool "enable to use port 443 for MQTT protocol"
```

Comment:
That's a good point, however I'm don't think the first config would actually be needed, as we always use TLS with AWS.

The difference here is, whether to use TLS ALPN extension or not. @ZYNQHRONIZE I assume that connecting over port 8883 w/o ALPN is still a valid method to connect to AWS? Is that correct? Perhaps we should keep it as a default, and add
```
config AWS_USE_TLS_ALPN
	bool "Use TLS ALPN extension when connecting to AWS"
```
And then, as Jukka suggested, another config for the port value. But instead, default to 8883, and optionally `default 443 if AWS_USE_TLS_ALPN`. WDYT?

We could also, instead of changing the default config file, add an overlay config that would enable ALPN support in the sample.

Reply:
yes if you want to connect with 8883, ALPN its not required. But if you want to use MQTT over port 443, it's require as shown in table below.  about if AWS_USE_TLS_ALPN ALPN was set then port will set to 443 by default, I think it's ok if we didn't have a plant to support AWS's Custom authentication.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/42055027/9623d0f7-1578-442b-8495-7d3d47c780dc)

Reply:
now my plan is if port is equal 443 and websocket is disabled then enable the x-amzn-mqtt-ca ALPN name if not then disable it.  and it will not interfered with ws WDYT. if both of your agree then I will proceed.

Reply:
Could be done that way as well, ok.

Reply:
Sounds ok to me.